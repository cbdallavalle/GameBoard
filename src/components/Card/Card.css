.Card {
  display: flex;
  flex-direction: row;
  margin: 10px auto;
  height: 260px;
  width: 100%;
  background-color: #ffffffcf;
  box-sizing: border-box;
  padding: 15px;
  font-family: 'Open Sans', sans-serif;
}

.Card h3 {
  font-weight: bold;
  text-align: center;
}

.Card img {
  margin-top: 10px;
  max-width: 180px;
}

.info-div {
  width: 50px;
  height: 50px;
  background-color: magenta;
}

.friends .game-info,
.games .game-info {
  width: 30%;
  line-height: 20px;
}

.search #game-title {
  width: 30%;
}

.search #game-description {
  width: 60%;
  line-height: 20px;
  padding: 0 40px 0 0;
}

.search #review-cont {
  display: none;
}

#game-description {
  padding: 0 5px;
  width: 40%;
  height: 240px;
  overflow-y: scroll;
}

#game-title {
  position: relative;
}

#delete, #add {
  width: 20px;
  position: absolute;
  bottom: -5px;
  left: -5px;
}

#delete, #add {
  cursor: pointer;
}

.search #delete,
.friends #delete {
  display: none;
}

.games #add,
.search #add {
  display: none;
}
