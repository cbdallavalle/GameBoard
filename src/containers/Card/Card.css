.Card {
  display: flex;
  flex-direction: row;
  position: relative;
  margin: 10px auto;
  height: 260px;
  width: 100%;
  background-color: #ffffffcf;
  box-sizing: border-box;
  padding: 15px;
  font-family: 'Open Sans', sans-serif; }

.games, .friends {
  animation: 0.7s cubic-bezier(0.29, 1.01, 0.45, 0.94) 0s 1 slideInFromLeft; }

@keyframes slideInFromLeft {
  0% {
    transform: translateX(-100%); }
  100% {
    transform: translateX(0); } }

.Card h3 {
  font-weight: bold;
  text-align: center; }

.Card img {
  margin-top: 10px;
  max-width: 180px; }

.info-div {
  width: 50px;
  height: 50px;
  background-color: magenta; }

.friends .game-info,
.games .game-info {
  width: 30%;
  line-height: 20px; }

.search #game-title {
  width: 30%; }

.search #game-description {
  width: 55%;
  line-height: 20px;
  padding: 0 40px 0 15px; }

.search #review-cont {
  display: none; }

#game-description {
  padding: 0 5px;
  width: 40%;
  height: 240px;
  overflow-y: scroll; }

#add,
#edit {
  position: absolute;
  top: 8px;
  right: 10px;
  width: 20px;
  height: 20px;
  border-radius: 10px;
  background-repeat: no-repeat;
  background-position: center; }
  #add:hover,
  #edit:hover {
    cursor: pointer; }

#add {
  background-image: url("../../assets/plus.svg"); }
  #add:hover {
    background-image: url("../../assets/plus-hover.svg"); }

#edit {
  background-image: url("../../assets/pencil.svg"); }
  #edit:hover {
    background-image: url("../../assets/pencil-hover.svg"); }

#delete, #add {
  cursor: pointer; }

#owned {
  width: 15px;
}

.search #edit,
.friends #edit {
  display: none; }

.games #add,
.search #add {
  display: none; }
