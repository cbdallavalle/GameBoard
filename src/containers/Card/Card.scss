.Card {
  display: flex;
  flex-direction: row;
  position: relative;
  margin: 10px auto;
  height: 260px;
  width: 100%;
  background-color: #ffffffcf;
  box-sizing: border-box;
  padding: 15px;
  font-family: 'Open Sans', sans-serif;
}

.Card h3 {
  font-weight: bold;
  text-align: center;
}

.Card img {
  margin-top: 10px;
  max-width: 180px;
}

.info-div {
  width: 50px;
  height: 50px;
  background-color: magenta;
}

.friends .game-info,
.games .game-info {
  width: 30%;
  line-height: 20px;
}

.search #game-title {
  width: 30%;
}

.search #game-description {
  width: 55%;
  line-height: 20px;
  padding: 0 40px 0 15px;
}

.search #review-cont {
  display: none;
}

#game-description {
  padding: 0 5px;
  width: 40%;
  height: 240px;
  overflow-y: scroll;
}

#add,
#edit {
  position: absolute;
  top: 8px;
  right: 10px;
  width: 20px;
  height: 20px;
  border-radius: 10px;
  background-repeat: no-repeat;
  background-position: center;
  &:hover {
    cursor: pointer;
  };
}

#add {
  background-image: url('../../assets/plus.svg');
  &:hover {
    background-image: url('../../assets/plus-hover.svg');
  };
}

#edit {
  background-image: url('../../assets/pencil.svg');
  &:hover {
    background-image: url('../../assets/pencil-hover.svg');
  }
}

#delete, #add {
  cursor: pointer;
}

.search #edit,
.friends #edit {
  display: none;
}

.games #add,
.search #add {
  display: none;
}
