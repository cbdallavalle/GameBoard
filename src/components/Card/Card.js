import React from 'react';
import './Card.css';

export const Card = () => {
  return (
    <article className="Card">
      <div className="game-info">
        <div className="info-div">
        </div>
        <h3>Above and Below</h3>
        <p>Awvkjs asjdfna awelsd alhwo ,s dlfoie als fljasef d,aslifhawelf sd,mnsld aleledld asjld. asdfd sljf e wls lksdf sld lkjf woe ij aldk asldkf d</p>
      </div>
      <div className="review-cont">
        <h3>Review</h3>
        <p>Awvkjs asjdfna awelsd alhwo ,s dlfoie als fljasef d,aslifhawelf sd,mnsld aleledld asjld. asdfd sljf e wls lksdf sld lkjf woe ij aldk asldkf d</p>
      </div>
    </article>
  )
}