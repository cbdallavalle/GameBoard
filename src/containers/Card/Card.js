import React, { Component } from 'react';
import PropTypes from 'prop-types';
import deleteIcon from '../../assets/error.svg';
import addIcon from '../../assets/plus.svg';
import editIcon from '../../assets/pencil.svg';
import check from '../../assets/success.svg';
import negative from '../../assets/negative.svg';
import { updateFavorites } from '../../actions';
import { Link, Route } from 'react-router-dom';
import EditGame from '../../components/EditGame/EditGame';

import { connect } from 'react-redux';
import { db } from '../../firebase';
import './Card.css';

export class Card extends Component {
  constructor() {
    super();
    this.state = {
      error: '',
      edit: false,
      review: {}
    };
  }

  componentDidMount = () => {
    this.getGameReview();
  }

  friendsName = () => {
    return this.props.type === 'friends' 
      ? this.props.friendName + ' played:' 
      : null;
  }

  handleEdit = () => {
    this.setState({ edit: !this.state.edit })
  }

  handleAdd = async () => {
    try {
      await db.doAddFavoriteData(this.props.user.uid, this.props.favorite);
      const favorites = await db.getFavorites(this.props.user.uid);
      this.props.updateFavorites(favorites);
    } catch (error) {
      this.setState({ error: error.message });
    }
  }

  getGameReview = async() => {
    try {
      const review = await db.doGetReviewData(this.props.user.uid, this.props.favorite.name);
      review && this.setState({ review });
    } catch (error) {
      this.setState({ error: error.message });
    }
  }

  determineOwned = () => {
    return this.state.review.owned === 'yes' 
    ? <p id="owned-p">owned <i class="far fa-check-circle"></i></p>
    : <p id="owned-p">played <i class="far fa-play-circle"></i></p>
  }

  determineDisplay = () => {
    const { description, name, thumbnail } = this.props.favorite;
    const { rating, review, timesPlayed } = this.state.review;

    return !this.state.edit ?
    <article className={`Card ${this.props.type}`}>
        <div className="game-info" id="game-title">
          <h3>{ this.friendsName() }</h3>
          <h3>{ name }</h3>
          <img src={ thumbnail } alt="game-icon" />
          { this.determineOwned() }
          <div
            id="edit"
            alt="edit a game in your collection"
            onClick={this.handleEdit}
          >
          </div>
          <div
            id="add"
            alt="edit a game in your collection"
            onClick={this.handleAdd} 
          >
          </div>
        </div>
        <div className="game-info" id="game-description">
          <p>{ description }</p>
        </div>
        <div className={`game-info`} id="review-cont">
          <h3><span>{rating}</span>/5</h3>
          <p>
            { review } 
          </p>
        </div>
      </article>
      :
      <EditGame favorite={this.props.favorite} handleEdit={this.handleEdit} />
  }

  render() {
    return (
      <div>
      { this.determineDisplay() }
      </div>
    ); 
  }
}

Card.propTypes = {
  user: PropTypes.object.isRequired,
  favorite: PropTypes.object.isRequired,
  friendName: PropTypes.string,
  type: PropTypes.string,
  display: PropTypes.string,
  updateFavorites: PropTypes.func.isRequired
};

export const mapStateToProps = state => ({
  user: state.user
});

export const mapDispatchToProps = dispatch => ({
  updateFavorites: favorites => dispatch(updateFavorites(favorites))
});

export default connect(mapStateToProps, mapDispatchToProps)(Card);