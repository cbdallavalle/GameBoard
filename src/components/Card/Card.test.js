import React from 'react';
import ReactDOM from 'react-dom';
import { Card } from './Card';
import { shallow } from 'enzyme';

describe("Card", () => {
  it("should match the snapshot", () => {
    const wrapper = shallow(
      <Card 
        user={{}}
        key={ 0 }
        favorite={ {} }
        friendName={ "friend" }
        type={ "games" }
      />
    )
    expect(wrapper).toMatchSnapshot();
  })
})