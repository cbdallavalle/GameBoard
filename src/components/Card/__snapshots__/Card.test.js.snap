// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="game-info"
>
          <div
                    className="info-div"
          />
          <h3>
                    Above and Below
          </h3>
          <p>
                    Awvkjs asjdfna awelsd alhwo ,s dlfoie als fljasef d,aslifhawelf sd,mnsld aleledld asjld. asdfd sljf e wls lksdf sld lkjf woe ij aldk asldkf d
          </p>
</div>,
        <div
          className="review-cont"
>
          <h3>
                    Review
          </h3>
          <p>
                    Awvkjs asjdfna awelsd alhwo ,s dlfoie als fljasef d,aslifhawelf sd,mnsld aleledld asjld. asdfd sljf e wls lksdf sld lkjf woe ij aldk asldkf d
          </p>
</div>,
      ],
      "className": "Card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="info-div"
/>,
            <h3>
              Above and Below
</h3>,
            <p>
              Awvkjs asjdfna awelsd alhwo ,s dlfoie als fljasef d,aslifhawelf sd,mnsld aleledld asjld. asdfd sljf e wls lksdf sld lkjf woe ij aldk asldkf d
</p>,
          ],
          "className": "game-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "info-div",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Above and Below",
            },
            "ref": null,
            "rendered": "Above and Below",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Awvkjs asjdfna awelsd alhwo ,s dlfoie als fljasef d,aslifhawelf sd,mnsld aleledld asjld. asdfd sljf e wls lksdf sld lkjf woe ij aldk asldkf d",
            },
            "ref": null,
            "rendered": "Awvkjs asjdfna awelsd alhwo ,s dlfoie als fljasef d,aslifhawelf sd,mnsld aleledld asjld. asdfd sljf e wls lksdf sld lkjf woe ij aldk asldkf d",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Review
</h3>,
            <p>
              Awvkjs asjdfna awelsd alhwo ,s dlfoie als fljasef d,aslifhawelf sd,mnsld aleledld asjld. asdfd sljf e wls lksdf sld lkjf woe ij aldk asldkf d
</p>,
          ],
          "className": "review-cont",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Review",
            },
            "ref": null,
            "rendered": "Review",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Awvkjs asjdfna awelsd alhwo ,s dlfoie als fljasef d,aslifhawelf sd,mnsld aleledld asjld. asdfd sljf e wls lksdf sld lkjf woe ij aldk asldkf d",
            },
            "ref": null,
            "rendered": "Awvkjs asjdfna awelsd alhwo ,s dlfoie als fljasef d,aslifhawelf sd,mnsld aleledld asjld. asdfd sljf e wls lksdf sld lkjf woe ij aldk asldkf d",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="game-info"
>
            <div
                        className="info-div"
            />
            <h3>
                        Above and Below
            </h3>
            <p>
                        Awvkjs asjdfna awelsd alhwo ,s dlfoie als fljasef d,aslifhawelf sd,mnsld aleledld asjld. asdfd sljf e wls lksdf sld lkjf woe ij aldk asldkf d
            </p>
</div>,
          <div
            className="review-cont"
>
            <h3>
                        Review
            </h3>
            <p>
                        Awvkjs asjdfna awelsd alhwo ,s dlfoie als fljasef d,aslifhawelf sd,mnsld aleledld asjld. asdfd sljf e wls lksdf sld lkjf woe ij aldk asldkf d
            </p>
</div>,
        ],
        "className": "Card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="info-div"
/>,
              <h3>
                Above and Below
</h3>,
              <p>
                Awvkjs asjdfna awelsd alhwo ,s dlfoie als fljasef d,aslifhawelf sd,mnsld aleledld asjld. asdfd sljf e wls lksdf sld lkjf woe ij aldk asldkf d
</p>,
            ],
            "className": "game-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "info-div",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Above and Below",
              },
              "ref": null,
              "rendered": "Above and Below",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Awvkjs asjdfna awelsd alhwo ,s dlfoie als fljasef d,aslifhawelf sd,mnsld aleledld asjld. asdfd sljf e wls lksdf sld lkjf woe ij aldk asldkf d",
              },
              "ref": null,
              "rendered": "Awvkjs asjdfna awelsd alhwo ,s dlfoie als fljasef d,aslifhawelf sd,mnsld aleledld asjld. asdfd sljf e wls lksdf sld lkjf woe ij aldk asldkf d",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Review
</h3>,
              <p>
                Awvkjs asjdfna awelsd alhwo ,s dlfoie als fljasef d,aslifhawelf sd,mnsld aleledld asjld. asdfd sljf e wls lksdf sld lkjf woe ij aldk asldkf d
</p>,
            ],
            "className": "review-cont",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Review",
              },
              "ref": null,
              "rendered": "Review",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Awvkjs asjdfna awelsd alhwo ,s dlfoie als fljasef d,aslifhawelf sd,mnsld aleledld asjld. asdfd sljf e wls lksdf sld lkjf woe ij aldk asldkf d",
              },
              "ref": null,
              "rendered": "Awvkjs asjdfna awelsd alhwo ,s dlfoie als fljasef d,aslifhawelf sd,mnsld aleledld asjld. asdfd sljf e wls lksdf sld lkjf woe ij aldk asldkf d",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
